sdk/python/tests/integration/registration/test_types.py

```diff
@@ -0,0 +1,273 @@
+import contextlib
+import tempfile
+import time
+import uuid
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Iterator, List, Tuple
+
+import pandas as pd
+import pytest
+from google.cloud import bigquery
+
+from feast import BigQuerySource
+from feast.data_source import DataSource
+from feast.entity import Entity
+from feast.feature import Feature
+from feast.feature_store import FeatureStore
+from feast.feature_view import FeatureView
+from feast.infra.online_stores.datastore import DatastoreOnlineStoreConfig
+from feast.repo_config import RepoConfig
+from feast.type_map import python_type_to_feast_value_type
+from feast.value_type import ValueType
+
+
+def get_entities_for_value_type(value_type: ValueType) -> List:
+    value_type_map = {
+        ValueType.INT32: [1, 2, 3, 4, 5],
+        ValueType.INT64: [1, 2, 3, 4, 5],
+        ValueType.FLOAT: [1.0, 2.0, 3.0, 4.0, 5.0],
+        ValueType.STRING: ["1", "2", "3", "4", "5"],
+    }
+    return value_type_map[value_type]
+
+
+def get_feature_values_for_dtype(dtype: str, is_list: bool) -> List:
+    if dtype is None:
+        return [0.1, None, 0.3, 4, 5]
+    # TODO: for int columns, consider having a better error when dealing with None values (pandas int dfs can't have na)
+    dtype_map = {
+        "int32": [1, 2, 3, 4, 5],
+        "int64": [1, 2, 3, 4, 5],
+        "float": [1.0, None, 3.0, 4.0, 5.0],
+        "string": ["1", None, "3", "4", "5"],
+        "bool": [True, False, None, True, False],
+    }
+    non_list_val = dtype_map[dtype]
+    # Duplicate the value once if this is a list
+    if is_list:
+        return [[n, n] if n is not None else None for n in non_list_val]
+    else:
+        return non_list_val
+
+
+def create_dataset(
```

Comment:
I think all the dataset creation functions can go in `tests/data/data_creator.py`. At least, that's what the structure of the first universal feature repo PR (#1734) suggests.

Reply:
Moved to universal repo. PTAL!